homeassistant/components/utility_meter/sensor.py

```diff
@@ -557,7 +557,7 @@ async def _async_reset_meter(self, event):
 
     async def async_reset_meter(self, entity_id):
         """Reset meter."""
-        if self._tariff_entity != entity_id:
+        if entity_id not in {self._tariff_entity, self.entity_id}:
```

Comment:
```suggestion
        if self._tariff != None and self._tariff_entity != entity_id:
```

Reply:
This way we don't change the current behaviour in which utility_meters with multiple tariffs are only reset through the `select.*` entity.

with your proposed change, you would enable individual tariff resets which would be an unintended change (IMHO)

Reply:
I see what you're saying, I'll test it out locally and push up an updated branch here soon.

Reply:
Tested and updated!